mk/Common.mk

```diff
@@ -610,14 +610,15 @@ vcs-unit-test:  vcs-run
 ###############################################################################
 # Build disassembler
 
-DPI_DASM_SRC = $(DPI_DASM_PKG)/dpi_dasm.cxx $(DPI_DASM_PKG)/spike/disasm.cc $(SPIKE_PKG)/disasm/regnames.cc
-DPI_DASM_OUT = $(DPI_DASM_PKG)/libdpi_dasm.so
+DPI_DASM_SRC    = $(DPI_DASM_PKG)/dpi_dasm.cxx $(DPI_DASM_PKG)/spike/disasm.cc $(SPIKE_PKG)/disasm/regnames.cc
+DPI_DASM_ARCH   = $(shell uname)$(shell getconf LONG_BIT)
+DPI_DASM_LIB    = $(DPI_DASM_PKG)/lib/$(DPI_DASM_ARCH)/libdpi_dasm.so
 DPI_DASM_CFLAGS = -shared -fPIC -std=c++11
-DPI_DASM_SVDPI_PATH ?= . -please_set_DPI_DASM_SVDPI_PATH_to_where_svdpi.h_is
-DPI_DASM_INC = -I $(DPI_DASM_PKG) -I $(DPI_DASM_SVDPI_PATH) -I $(SPIKE_PKG)/riscv -I $(SPIKE_PKG)/softfloat
+DPI_DASM_INC    = -I$(DPI_DASM_PKG) -I$(DPI_INCLUDE) -I$(SPIKE_PKG)/riscv -I$(SPIKE_PKG)/softfloat
+DPI_DASM_CXX    = g++
```

Comment:
There is a reason I used `c++` and not `g++`, and an example is shown in the attached screenshot.
However, I think the right choice is `g++` and _if_ there are any troubles (which our future setup should be able to catch) then fix it from there.
I just confirmed that isacov itself appears to work fine with the .so you provided and even after rebuilding the .so.

![image](https://user-images.githubusercontent.com/75431602/115891349-482d4080-a456-11eb-997b-ec718e6afe9b.png)

Reply:
My guess is perhaps this is side-effect of the global string issue above.  At this point we generally universally use GNU tools to build both applications and firmware code so for sake of consistency I generally point C builds to gcc and C++ to g++ by default and the user can retarget the make variables as needed.